File: java/mediator/exercise/TravelAirplane.java
package mediator.exercise;

public class TravelAirplane implements Airplane {

    private AirTrafficControlMediator mediator;

    public TravelAirplane(AirTrafficControlMediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public String getType() {
        return "Travel";
    }

    @Override
    public void requestTakeoff() {
        mediator.requestTakeOff(this);
    }

    @Override
    public void requestLanding() {
        mediator.requestLanding(this);
    }

    @Override
    public void notifyAirTrafficControl(String message) {
        System.out.println("Travel Airplane: " + message);
    }
}

File: C#/Mediator/Exercise/TravelAirplane.cs
namespace Mediator.Exercise;

public sealed class TravelAirplane : IAirplane
{
    public string Type { get; } = "Travel";

    private readonly IAirTrafficControlMediator _airTrafficControlMediator;

    public TravelAirplane(IAirTrafficControlMediator airTrafficControl)
        => _airTrafficControlMediator = airTrafficControl;

    public void NotifyAirTrafficControl(string message)
        => Console.WriteLine($"Travel Airplane: {message}");

    public void RequestLanding()
        => _airTrafficControlMediator.RequestLanding(this);

    public void RequestTakeoff()
        => _airTrafficControlMediator.RequestTakeOff(this);
}
